inventory_level.go

```diff
@@ -0,0 +1,76 @@
+package shopify
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+//InventoryLevel struct for Shopify inventory_level
+type InventoryLevel struct {
+	InventoryItemID     int64  `json:"inventory_item_id"`
+	LocationID          int64  `json:"location_id"`
+	Available           int64  `json:"available,omitempty"`
+	AvailableAdjustment int64  `json:"available_adjustment,omitempty"`
+	UpdatedAt           string `json:"updated_at,omitempty"`
+
+	api *API
+}
+
+//Connect connect an inventory item to a location
+func (obj *InventoryLevel) Connect() error {
+	endpoint := fmt.Sprintf("/admin/inventory_levels/connect.json")
+	return requestInvLevel(endpoint, "POST", obj)
+}
+
+//Set set an inventory level for a variant w. location id
+func (obj *InventoryLevel) Set() error {
+	endpoint := fmt.Sprintf("/admin/inventory_levels/set.json")
+	return requestInvLevel(endpoint, "POST", obj)
+}
+
+//Adjust adjust an inventory level for a inventory item w. location id
+func (obj *InventoryLevel) Adjust() error {
+	endpoint := fmt.Sprintf("/admin/inventory_levels/adjust.json")
+	return requestInvLevel(endpoint, "POST", obj)
+}
+
+//Delete delete an inventory level for a inventory item w. location id
+func (obj *InventoryLevel) Delete() error {
+	endpoint := fmt.Sprintf("/admin/inventory_levels.json?inventory_item_id=%&location_id=%", obj.InventoryItemID, obj.LocationID)
+	return requestInvLevel(endpoint, "DELETE", obj)
+}
+
+func requestInvLevel(endpoint, method string, obj *InventoryLevel) error {
+	expectedStatus := 200
+	var buf bytes.Buffer
+	body := map[string]*InventoryLevel{
```

Comment:
The body for most methods consuming this function just requires the `InventoryLevel` object as the full payload, rather than it being an object with a key "inventory_levels" with the object, e.g.

```
POST /admin/inventory_levels/set.json

{
  "location_id": 905684977,
  "inventory_item_id": 808950810,
  "available": 42
}
```